framework/include/utils/InputParameterWarehouse.h

```diff
@@ -127,4 +136,61 @@ class InputParameterWarehouse : public Warehouse<InputParameters>
   friend class SolidMaterialProperties;
 };
 
+
+template<typename T>
+ControllableParameter<T>
+InputParameterWarehouse::getControllableParameter(const MooseObjectParameterName & input, bool mark_as_controlled /*=false*/)
+{
+
+  // The ControllableParameter object to return
+  ControllableParameter<T> output;
+
+  // Vector of desired parameters
+  std::vector<MooseObjectParameterName> params(1, input);
+  std::map<MooseObjectParameterName, std::vector<MooseObjectParameterName> >::const_iterator link_it = _input_parameter_links.find(input);
+  if (link_it != _input_parameter_links.end())
+    params.insert(params.end(), link_it->second.begin(), link_it->second.end());
+
+  // Parameter object iterator
+  std::multimap<MooseObjectName, MooseSharedPointer<InputParameters> >::const_iterator iter;
+
+  // Loop over all threads
+  for (THREAD_ID tid = 0; tid < libMesh::n_threads(); ++tid)
+  {
+    // Loop over all InputParameter objects
+    for (iter = _input_parameters[tid].begin(); iter != _input_parameters[tid].end(); ++iter)
+    {
+      // Loop of all desired params
+      for (std::vector<MooseObjectParameterName>::iterator it = params.begin(); it != params.end(); ++it)
```

Comment:
Also, it looks like we are doing a linear search (even though these are probably very small vectors). This probably isn't a performance issue, I'm just pointing it out in case there's a more elegant way we're missing.